datasets/speechocean762/speechocean762.py

```diff
@@ -0,0 +1,174 @@
+                "age": datasets.Value("int32"),
+                "gender": datasets.Value("string"),
+                "text": datasets.Value("string"),
+                "audio": datasets.Audio(sampling_rate=_SAMPLE_RATE),
+                "accuracy": datasets.Value("float32"),
+                "completeness": datasets.Value("float32"),
+                "fluency": datasets.Value("float32"),
+                "prosodic": datasets.Value("float32"),
+                "total": datasets.Value("float32"),
+                "words": datasets.Sequence(
+                    {
+                        "text": datasets.Value("string"),
+                        "accuracy": datasets.Value("float32"),
+                        "stress": datasets.Value("float32"),
+                        "total": datasets.Value("float32"),
+                        "phones": datasets.Sequence(datasets.Value("string")),
+                        "phones-accuracy": datasets.Sequence(datasets.Value("float32")),
+                        "mispronunciations": datasets.Sequence(
+                            {
+                                "index": datasets.Value("int32"),
+                                "canonical-phone": datasets.Value("string"),
+                                "pronounced-phone": datasets.Value("string"),
+                            }
+                        ),
+                    }
+                ),
+                "original_full_path": datasets.Value("string"),  # relative path to full audio in original data dirs
+            }
+        )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        archive_path = dl_manager.download(_DL_URL)
+        local_extracted_archive = dl_manager.extract(archive_path) if not dl_manager.is_streaming else {}
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"local_extracted_archive": local_extracted_archive, "split": "train"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"local_extracted_archive": local_extracted_archive, "split": "test"},
+            ),
+        ]
+
+    def _generate_examples(self, local_extracted_archive, split):
+        root_path = Path(local_extracted_archive) / "speechocean762"
+        meta_path = root_path / f"{split}.json"
+
+        with open(meta_path, encoding="utf-8") as f:
+            info = json.load(f)
+            for utt_id in info:
+                audio = info[utt_id]
+                audio["id"] = utt_id
+                audio["original_full_path"] = f"{root_path}/WAVE/SPEAKER{audio['speaker']}/{utt_id}.WAV"
+                audio_data, _ = librosa.load(audio["original_full_path"], sr=_SAMPLE_RATE)
```

Comment:
I don't think we should load the audio data here - this is done automatically when the feature "audio" is being called. 

Could we maybe instead do:
```suggestion
                audio_bytes = audio["original_full_path"].read()
                ...
                audio["audio"] = {"path": audio["original_full_path"], "bytes": audio_bytes}
```

Reply:
Thanks for your kind suggestion. I think `audio["original_full_path"].read()` would not work here because `audio["original_full_path"]` is a str without the `read()` method:

```python
# Line 170 of datasets/speechocean762/speechocean762.py
audio["original_full_path"] = f"{root_path}/WAVE/SPEAKER{audio['speaker']}/{utt_id}.WAV"
```

I didn't find related example codes in `datasets/librispeech_asr/librispeech_asr.py`. So I'm not sure how to make the feature "audio" with `read()` :flushed:

Reply:
Once you iterate over the files of the TAR archive as per @lhoestq's suggestion you will be able to `.read()` the `audio_file`!